pkg/tsdb/cloudmonitoring/cloudmonitoring.go

```diff
@@ -66,145 +71,213 @@ const (
 	perSeriesAlignerDefault   string = "ALIGN_MEAN"
 )
 
-func ProvideService(cfg *setting.Cfg, pluginManager plugins.Manager, httpClientProvider httpclient.Provider) *Service {
-	return &Service{
-		PluginManager:      pluginManager,
-		HTTPClientProvider: httpClientProvider,
-		Cfg:                cfg,
+func ProvideService(cfg *setting.Cfg, httpClientProvider httpclient.Provider, pluginManager plugins.Manager,
+	backendPluginManager backendplugin.Manager, dataSourceCache datasources.CacheService) *Service {
+	s := &Service{
+		pluginManager:        pluginManager,
+		backendPluginManager: backendPluginManager,
+		httpClientProvider:   httpClientProvider,
+		dataSourceCache:      dataSourceCache,
+		cfg:                  cfg,
+		im:                   datasource.NewInstanceManager(newInstanceSettings(httpClientProvider)),
+	}
+
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: s,
+	})
+
+	if err := s.backendPluginManager.Register("stackdriver", factory); err != nil {
+		slog.Error("Failed to register plugin", "error", err)
 	}
+	return s
 }
 
 type Service struct {
-	PluginManager      plugins.Manager
-	HTTPClientProvider httpclient.Provider
-	Cfg                *setting.Cfg
+	pluginManager        plugins.Manager
+	backendPluginManager backendplugin.Manager
+	dataSourceCache      datasources.CacheService
+	httpClientProvider   httpclient.Provider
+	cfg                  *setting.Cfg
+	im                   instancemgmt.InstanceManager
 }
 
-// Executor executes queries for the CloudMonitoring datasource.
-type Executor struct {
-	httpClient    *http.Client
-	dsInfo        *models.DataSource
-	pluginManager plugins.Manager
-	cfg           *setting.Cfg
+type QueryModel struct {
+	Type string `json:"type"`
 }
 
-// NewExecutor returns an Executor.
-//nolint: staticcheck // plugins.DataPlugin deprecated
-func (s *Service) NewExecutor(dsInfo *models.DataSource) (plugins.DataPlugin, error) {
-	httpClient, err := dsInfo.GetHTTPClient(s.HTTPClientProvider)
-	if err != nil {
-		return nil, err
-	}
+type datasourceInfo struct {
+	id                 int64
+	updated            time.Time
+	url                string
+	authenticationType string
+	defaultProject     string
+	client             *http.Client
 
-	return &Executor{
-		httpClient:    httpClient,
-		dsInfo:        dsInfo,
-		pluginManager: s.PluginManager,
-		cfg:           s.Cfg,
-	}, nil
+	jsonData                map[string]interface{}
+	decryptedSecureJSONData map[string]string
+}
+
+func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		var jsonData map[string]interface{}
```

Comment:
Yeah I was hoping to do this but unfortunately because of the way the plugin proxy expects the json data, it's simpler to do it this way. If we introduce a struct then we'll have to unmarshal twice or construct a `map[string]interface{}` manually